player/multiview/js/script.js

```diff
@@ -0,0 +1,271 @@
+    const newPrimaryPlayer = findPlayerForSource(activeSources[0]);
+    setPrimaryPlayer(newPrimaryPlayer, activeSources[0]);
+  }
+  const player = findPlayerForSource(source);
+  player?.unload().then(() => {
+    unusedPlayers.push(player);
+  });
+}
+
+function updateGrid() {
+  const grid = document.getElementById('grid');
+
+  // Clear existing tiles and classes
+  grid.innerHTML = '';
+  grid.className = 'grid';
+
+  // Add tiles dynamically
+  for (let i = 0; i < activeSources.length; i++) {
+    const source = activeSources[i];
+    const player = getPlayerInstance(playerConfig, source);
+
+    const tile = player.getContainer();
+    tile.title = source.title;
+
+    if (primarySource === null) {
+      setPrimaryPlayer(player, source);
+    }
+
+    grid.appendChild(tile);
+  }
+
+  // Set grid layout using CSS classes
+  grid.classList.add(`tile-count-${activeSources.length}`);
+}
+
+function getPlayerId(player) {
+  return allPlayers.indexOf(player);
+}
+
+function getControlBar(player) {
+  const playerId = getPlayerId(player);
+  return controlBarsContainer.querySelector('.bmpui-ui-controlbar[data-player-id="' + playerId + '"]');
+}
+
+function getPlayerInstance(playerConfig, source) {
+  let player = findPlayerForSource(source);
+  if (player) {
+    // An active player instance already exists for this source
+    return player;
+  }
+
+  player = unusedPlayers.shift();
+  if (player) {
+    // Re-use one of the unused player instances
+    player.load(source);
+    return player;
+  }
+
+  // Create a new player instance
+  const container = createPlayerTile();
+  const newPlayer = new bitmovin.player.Player(container, playerConfig);
+  container.addEventListener('click', event => onTileClicked(container, newPlayer, event), true);
+
+  newPlayer.load(source).then(() => {
+    // Move the control bar to the controlbars-container to be full-width
+    // This lets us simulate a common control bar for all players
+
+    const controlBar = container.getElementsByClassName('bmpui-ui-controlbar')[0];
+    controlBar.setAttribute('data-player-id', getPlayerId(newPlayer).toString());
+    if (source === primarySource) {
+      controlBar.classList.add('active');
+    }
+
+    controlBarsContainer.appendChild(controlBar);
+  });
+  allPlayers.push(newPlayer);
+
+  return newPlayer;
+}
+
+function findPlayerForSource(source) {
+  return allPlayers.find(player => player.getSource() === source);
+}
+
+function setPrimaryPlayer(newPrimaryPlayer, newPrimarySource) {
+  if (newPrimarySource === primarySource || newPrimaryPlayer === null) {
+    return;
+  }
+
+  const container = document.getElementById('player-container');
+
+  // Remove active classes from the previous primary player
+  container.querySelector('.bmpui-ui-controlbar.active')?.classList.remove('active');
+  container.querySelector('.tile.primary')?.classList.remove('primary');
+
+  if (newPrimaryPlayer == null) {
+    primarySource = null;
+    return;
+  }
```

Comment:
you already checked for null at the start of the function

Reply:
nice catch! Actually I needed to keep this one and remove the one at the start, otherwise we'll run into an exception when the last source gets disabled.

https://github.com/bitmovin/demos/pull/124/commits/067e269cad6c270b5670e1615c4cf27c9a6ec0fd